### PR TITLE
PARQUET-1427: [C++] Incorporate Parquet CLI tools and example executables into monorepo codebase

### PR DESCRIPTION
**Pull request by wesm**

### CODE REVIEW

**Comment by wesm** [comment]
cc @xhochy @majetideepak. Like the other merge patch, this is to be merged with `--ff-only` like the other patch. Please don't rebase unless you use the cherry-pick-with-committer approach (https://github.com/dingram/git-scripts/blob/master/scripts/git-cherry-pick-with-committer)

**Review by xhochy — APPROVED** [review]
+1, merge on green

**Comment by wesm** [comment]
Rebased and fixed the parquet-arrow example. Will merge once the build is looking good

**Pull request closed without merging**
